src/trans.rs

```diff
@@ -914,9 +964,16 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
     fn trans_lval(&mut self, lvalue: &Lvalue<'tcx>) -> BinaryenLvalue {
         let i = match *lvalue {
-            Lvalue::Arg(i) => i,
-            Lvalue::Var(i) => self.mir.arg_decls.len() as u32 + i,
+            Lvalue::Arg(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
+                i
+            },
+            Lvalue::Var(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
+                self.mir.arg_decls.len() as u32 + i
+            }
             Lvalue::Temp(i) => {
+                let i: u32 = unsafe { mem::transmute(i) };
```

Comment:
Never, *ever* use `transmute` except as a last resort. These newtypes have an API for getting their index integer.
Besides, this computation has been replaced with `local_index` or something like that, also used by `rustc_trans`.